sky/packages/sky/lib/src/fn3/scaffold.dart

```diff
@@ -0,0 +1,252 @@
+    for (ScaffoldSlots slot in ScaffoldSlots.values.reversed) {
+      RenderBox box = _slots[slot];
+      if (box != null) {
+        assert(box.parentData is BoxParentData);
+        if (box.hitTest(result, position: (position - box.parentData.position).toPoint()))
+          return;
+      }
+    }
+  }
+
+  String debugDescribeChildren(String prefix) {
+    return _slots.keys.map((slot) => '${prefix}${slot}: ${_slots[slot].toStringDeep(prefix)}').join();
+  }
+}
+
+class Scaffold extends RenderObjectWidget {
+  Scaffold({
+    Key key,
+    Widget body,
+    Widget statusBar,
+    Widget toolbar,
+    Widget snackBar,
+    Widget floatingActionButton,
+    Widget drawer
+  }) : super(key: key) {
+    _children[ScaffoldSlots.body] = body;
+    _children[ScaffoldSlots.statusBar] = statusBar;
+    _children[ScaffoldSlots.toolbar] = toolbar;
+    _children[ScaffoldSlots.snackBar] = snackBar;
+    _children[ScaffoldSlots.floatingActionButton] = floatingActionButton;
+    _children[ScaffoldSlots.drawer] = drawer;
+  }
+
+  final Map<ScaffoldSlots, Widget> _children = new Map<ScaffoldSlots, Widget>();
+
+  RenderScaffold createRenderObject() => new RenderScaffold();
+
+  ScaffoldElement createElement() => new ScaffoldElement(this);
+}
+
+class ScaffoldElement extends RenderObjectElement<Scaffold> {
+  ScaffoldElement(Scaffold widget) : super(widget);
+
+  Map<ScaffoldSlots, Element> _children;
+
+  void visitChildren(ElementVisitor visitor) {
+    for (ScaffoldSlots slot in ScaffoldSlots.values) {
+      Element element = _children[slot];
+      if (element != null)
+        visitor(element);
+    }
+  }
+
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    _children = new Map<ScaffoldSlots, Element>();
+    for (ScaffoldSlots slot in ScaffoldSlots.values) {
+      _children[slot] = widget._children[slot].createElement()
+        ..mount(this, slot);
+    }
+  }
+
+  void update(Scaffold newWidget) {
+    super.update(newWidget);
+    assert(widget == newWidget);
+    for (ScaffoldSlots slot in ScaffoldSlots.values) {
+      _children[slot] = updateChild(_children[slot], widget._children[slot], slot);
+      assert((_children[slot] == null) == (widget._children[slot] == null));
+    }
+  }
+
+  void insertChildRenderObject(RenderObject child, ScaffoldSlots slot) {
+    final renderObject = this.renderObject; // TODO(ianh): Remove this once the analyzer is cleverer
+    assert(renderObject is RenderScaffold);
+    renderObject[slot] = child;
+    assert(renderObject == this.renderObject); // TODO(ianh): Remove this once the analyzer is cleverer
+  }
+
+  void moveChildRenderObject(RenderObject child, dynamic slot) {
+    removeChildRenderObject(child);
+    insertChildRenderObject(child, slot);
+  }
+
+  void removeChildRenderObject(RenderObject child) {
+    final renderObject = this.renderObject; // TODO(ianh): Remove this once the analyzer is cleverer
```

Comment:
you should just override renderObject to return a RenderScaffold
```dart
   RenderScaffold renderObject => super.renderObject;
```
then this function becomes:
```dart
assert(renderObject == child.parent);
renderObject.remove(child);
```